ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/PDEFormEditor.java

```diff
@@ -480,11 +480,11 @@ public void doSaveAs() {
 			// Ignore
 		} catch (Exception e) {
 			String title = PDEUIMessages.PDEFormEditor_errorTitleProblemSaveAs;
-			String message = PDEUIMessages.PDEFormEditor_errorMessageSaveNotCompleted;
+			StringBuilder message = new StringBuilder().append(PDEUIMessages.PDEFormEditor_errorMessageSaveNotCompleted);
```

Comment:
Odd, I would have thought that this would be more performant (and would generated less byte code) with one less method call:
```
new StringBuilder(PDEUIMessages.PDEFormEditor_errorMessageSaveNotCompleted);
```

Reply:
JDT clean-ups do not generated perfect code, could be enhanced with a follow-up commit. Mixing automatic clean-ups with manual adjustments should IMHO avoided

Reply:
It just seems ironic that a performance cleanup generates glaringly suboptimal code, both in terms of performance and verbosity.  I suppose it's because there are more append overloads than constructor overloads, so JDT would need to be smarter and more careful when generating a constructor call...

Reply:
@vogella you can just push another commit to your branch as well for the manual work (or revert it + bugreport to JDT).

i also don't see why this is more "performant" (regardless of if one needs to optimize performance for the error case). I think if you really care about performance the code can be rewritten in one constant logging and one with exception like this:

```
if (e.getMessage() == null) {
    PDEPlugin.logException(e, title, PDEUIMessages.PDEFormEditor_errorTitleProblemSaveAs);
} else {
    PDEPlugin.logException(e, title, PDEUIMessages.PDEFormEditor_errorTitleProblemSaveAs+ '  ' + e.getMessage());
```
}

Reply:
> Odd, I would have thought that this would be more performant (and would generated less byte code) with one less method call:

Even that is wasteful as for the case of message= null it created a `StringBuilder` from a constant to convert it into a string right after that...

Reply:
Indeed, my comment was just from looking at the one line in isolation of all context...